data/types/base/base.go

```diff
@@ -17,42 +17,47 @@ import (
-
-	Annotations      *[]interface{} `json:"annotations,omitempty" bson:"annotations,omitempty"`           // PARSE
-	ClockDriftOffset *int           `json:"clockDriftOffset,omitempty" bson:"clockDriftOffset,omitempty"` // PARSE
-	ConversionOffset *int           `json:"conversionOffset,omitempty" bson:"conversionOffset,omitempty"` // PARSE
-	DeviceID         *string        `json:"deviceId,omitempty" bson:"deviceId,omitempty"`                 // PARSE
-	DeviceTime       *string        `json:"deviceTime,omitempty" bson:"deviceTime,omitempty"`             // PARSE
-	Payload          *interface{}   `json:"payload,omitempty" bson:"payload,omitempty"`                   // PARSE
-	Time             *string        `json:"time,omitempty" bson:"time,omitempty"`                         // PARSE
-	TimezoneOffset   *int           `json:"timezoneOffset,omitempty" bson:"timezoneOffset,omitempty"`     // PARSE
-	Version          *int           `json:"-" bson:"_version,omitempty"`                                  // SET
+	Active        bool   `json:"-" bson:"_active"`
+	CreatedTime   string `json:"createdTime,omitempty" bson:"createdTime,omitempty"`
+	GroupID       string `json:"-" bson:"_groupId,omitempty"`
+	GUID          string `json:"guid,omitempty" bson:"guid,omitempty"`
+	ID            string `json:"id,omitempty" bson:"id,omitempty"`
+	SchemaVersion int    `json:"-" bson:"_schemaVersion,omitempty"`
+	Type          string `json:"type,omitempty" bson:"type,omitempty"`
+	UploadID      string `json:"uploadId,omitempty" bson:"uploadId,omitempty"`
+	UserID        string `json:"-" bson:"_userId,omitempty"`
+	Version       int    `json:"-" bson:"_version,omitempty"`
+
+	Annotations      *[]interface{} `json:"annotations,omitempty" bson:"annotations,omitempty"`
+	ClockDriftOffset *int           `json:"clockDriftOffset,omitempty" bson:"clockDriftOffset,omitempty"`
+	ConversionOffset *int           `json:"conversionOffset,omitempty" bson:"conversionOffset,omitempty"`
+	DeviceID         *string        `json:"deviceId,omitempty" bson:"deviceId,omitempty"`
+	DeviceTime       *string        `json:"deviceTime,omitempty" bson:"deviceTime,omitempty"`
+	Payload          *interface{}   `json:"payload,omitempty" bson:"payload,omitempty"`
+	Time             *string        `json:"time,omitempty" bson:"time,omitempty"`
+	TimezoneOffset   *int           `json:"timezoneOffset,omitempty" bson:"timezoneOffset,omitempty"`
 }
 
 type Meta struct {
 	Type string `json:"type,omitempty"`
 }
 
-func New(Type string) (*Base, error) {
-	if Type == "" {
+func New(baseType string) (*Base, error) {
+	if baseType == "" {
 		return nil, app.Error("base", "type is missing")
 	}
 
 	return &Base{
 		Active:        false,
-		CreatedTime:   time.Now().Format(time.RFC3339),
+		CreatedTime:   time.Now().UTC().Format(time.RFC3339),
+		GUID:          app.NewUUID(),
```

Comment:
I added GUID back in because it is listed in the data model docs. The uploader has been adding it for a long while now. We are now generating it in the backend (and the uploader no longer sends it). The ID is definitely for backwards compatibility. At some point we should settle on one or the other and deprecate one of them.